src/backend/commands/explain_gp.c

```diff
@@ -1492,6 +1493,38 @@ cdbexplain_showExecStatsBegin(struct QueryDesc *queryDesc,
 	return ctx;
 }								/* cdbexplain_showExecStatsBegin */
 
+/*
+ * cdbexplain_showStatCtxFree
+ *	  Release memory allocated for CdbExplain_ShowStatCtx structure and its
+ *	  internals.
+ */
+void
+cdbexplain_showStatCtxFree(struct CdbExplain_ShowStatCtx *ctx)
```

Comment:
How about instead this use `MemoryContextSwitchTo(queryDesc->estate->es_query_cxt)` in `cdbexplain_showExecStatsBegin`?